### DISCUSSION TITLE
Feature Request: Footnotes

### DISCUSSION THREAD

**Discussion by thierna** [marp-team/marp#150] [Ideas] [answered]
Thanks for this nice tool to use markdown for slides. As I work in scientific research, it would be really great to have footnote support included in marp. Is there any plan for doing this?

**Comment by yhatt** [comment, marked as answer]
No. Marp respects CommonMark specification for getting compatibility with other Markdown processors. The footnote syntax is out of CommonMark spec so we are not going to work for getting footnotes.

[Marp syntax can extend by using plugins](https://marpit.marp.app/usage?id=extend-marpit-by-plugins), but unfortunately there is no footnote plugins suited for Marp slides at this time. We hope community makes a footnote plugin for Marp.

### Related issues

- https://github.com/yhatt/marp/issues/2
- https://github.com/marp-team/marp-cli/issues/226
- https://github.com/marp-team/marp-cli/issues/300

**Comment by Juanvvc** [comment]
If you think it is useful, I'm using custom styled blockquotes as footnotes.

```
<style>
blockquote {
    border-top: 0.1em dashed #555;
    font-size: 60%;
    margin-top: auto;
}
</style>

# Title

Text text text

> This is a footnote
```

An example in my slides (Spanish): https://juanvvc.github.io/crypto/slides/03-simetrica.html#5

**Comment by kaiiam** [comment]
The following, modified from [this stackoverflow](https://stackoverflow.com/questions/66964/how-do-i-create-a-link-to-a-footnote-in-html) works for me using marp for vs code on a mac:

```
# Footenote

This is main body of my content. I have a footnote link for this line <a href="#footnote-1">[1]</a>. Then, I have some more content. Some of it is interesting and it has some footnotes as well <a href="#footnote-2">[2]</a>.

Here is another paragraph of text with a third footnote <a href="#footnote-3">[3]</a>.
<br></br>


<h4></h4>
<sub><sub><sub><ol>
    <li id="footnote-1">Here is my first footnote.</li>
    <li id="footnote-2">Another footnote.</li>
    <li id="footnote-3">Footnote three.</li>
</ol></sub></sub></sub>

``` 
renders as

<img width="700" alt="image" src="https://github.com/marp-team/marp/assets/12255688/5f0bf770-5c03-4bf6-aee8-d794ef59e5d3">

**Comment by kaorukobo** [comment]
I confirmed that marp-cli 4.2.0 and [markdown-it-footnote](https://github.com/markdown-it/markdown-it-footnote) 4.0.0 work together, while I'm still unsure why the *markdown-it* plugin works with `marp.use()`:

```bash
# add markdown-it-footnote dependency
pnpm i markdown-it-footnote

# create marp-cli config file.
cat > marp.config.mjs <<'JS'
import markdownItFootnote from 'markdown-it-footnote';

export default {
  engine: ({ marp }) => marp.use(markdownItFootnote),
};
JS

# convert markdown with footnote syntax
pnpx @marp-team/marp-cli@latest foo.md
```

**Reply by yhatt** [reply]
> while I'm still unsure why the _markdown-it_ plugin works with marp.use():

Marp and its base framework [Marpit](https://marpit.marp.app/) have [a plugin architecture that is compatible with markdown-it plugins][marpit-api]. This makes it easy for users to extend the Markdown syntax, powered by existing [markdown-it plugin ecosystem](https://www.npmjs.com/search?q=keywords:markdown-it-plugin) (if compatible). Refer to [Marpit API documentation][marpit-api] for more details.

[marpit-api]: https://marpit-api.marp.app/

Formerly `markdown-it-footnote` had reported that is incompatible when using `footer` directive in Marp (https://github.com/marp-team/marp-cli/issues/300#issuecomment-728113607). But currently, it seems like there are no problems of blocking render.

`markdown-it-footnote` will generate the footnotes section at the end of document, i.e. the last slide in Marp, so note that it will not generate footnotes per slides such as [Deckset](https://docs.deckset.com/English.lproj/Formatting/11-footnotes.html).
